src/core/exec-invoke.c

```diff
@@ -5423,6 +5462,36 @@ int exec_invoke(
         if (r < 0)
                 return r;
 
+        if (context->private_bpf) {
+                _cleanup_close_ int fs_fd = -EBADF, mnt_fd = -EBADF, bpffs_fd = -EBADF;
+
+                fs_fd = fsopen("bpf", 0);
+                if (fs_fd < 0)
+                        return fs_fd;
+
+                r = send_one_fd(token_fds[0], fs_fd, 0);
+                if (r < 0)
+                        return r;
+
+                mnt_fd = receive_one_fd(token_fds[0], 0);
+                if (mnt_fd < 0)
+                        return fs_fd;
+
+                bpffs_fd = openat(mnt_fd, ".", O_RDONLY);
+                if (bpffs_fd < 0)
+                        return bpffs_fd;
+
+                r = move_mount(mnt_fd, "", AT_FDCWD, "/sys/fs/bpf", MOVE_MOUNT_F_EMPTY_PATH);
+                if (r < 0)
+                        return r;
+
+                r = wait_for_terminate_and_check("(bpffs)", TAKE_PID(bpftoken_pid), 0);
+                if (r < 0)
+                        return r;
+                if (r != EXIT_SUCCESS) /* If something strange happened with the child, let's consider this fatal, too */
+                        return -EIO;
```

Comment:
Should be moved before receive_one_fd(), otherwise send_one_fd() may stuck on failure.